rfc/rfc017-distributed-network-grpc-v2.md

```diff
@@ -45,4 +45,11 @@ As specified by RFC015, the node MUST authenticate the peer's node DID as follow
 1. Resolve peer's node DID to its corresponding DID document.
 2. Assert that one of the `NutsComm` endpoint's host matches (one of) the `dNSName` SANs in the peer's TLS client certificate.
 
-## X. Private Transactions
\ No newline at end of file
+## X. Private Transactions
+
+## X. Service Discovery
+
+Although a (or multiple) trusted bootstrap node is required for initially connecting to the network, nodes SHOULD discover new peers by searching for `NutsComm` endpoints in the received DID documents.
+This SHOULD be done after the initial DAG sync is completed, to avoid connecting to non-existing endpoints resolved from older versions of DID documents.
```

Comment:
NutsComm services with references to other DID Documents MUST be ignored. Only consider concrete endpoints.